src/question/order_select/builder.rs

```diff
@@ -0,0 +1,277 @@
+        self
+    }
+
+    /// Whether to wrap around when user gets to the last element.
+    ///
+    /// If `should_loop` is not set, it will default to `true`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use requestty::Question;
+    ///
+    /// let order_select = Question::order_select("cheese")
+    ///     .should_loop(false)
+    ///     .build();
+    /// ```
+    pub fn should_loop(mut self, should_loop: bool) -> Self {
+        self.order_select.choices.set_should_loop(should_loop);
+        self
+    }
+
+    /// Extends the given iterator of [`Choice`]s
+    ///
+    /// The choices are [`String`]s and can be multiline.
+    ///
+    /// [`Choice`]: crate::question::Choice
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use requestty::Question;
+    ///
+    /// let order_select = Question::order_select("hamburger")
+    ///     //...
+    ///     .choices(vec![
+    ///         "Salad",
+    ///         "Cheddar",
+    ///         "Cheese",
+    ///     ])
+    ///     //...
+    ///     .build();
+    /// ```
+    pub fn choices<I, T>(mut self, choices: I) -> Self
+    where
+        T: Into<String>,
+        I: IntoIterator<Item = T>,
+    {
+        let len = self.order_select.choices.choices.len();
+
+        self.order_select.choices.choices.extend(
+            choices
+                .into_iter()
+                .enumerate()
+                .map(|(i, c)|
+                    OrderSelectItem { 
+                        initial_index:len + i, 
+                        text: Text::new(c.into())
+                    }
+                ),
+        );
+        self
+    }
+
+    crate::impl_filter_builder! {
+        /// # Examples
+        ///
+        /// ```
+        /// use requestty::Question;
+        ///
+        /// let order_select = Question::order_select("evil_home_tasks")
+        ///     //...
+        ///     .filter(|mut tasks, previous_answers| {
+        ///         tasks.rotate_left(1);
+        ///         tasks
+        ///     })
+        ///     //...
+        ///     .build();
+        /// ```
+        Vec<OrderSelectItem>; order_select
+    }
+
+    crate::impl_validate_builder! {
+        /// NOTE: The boolean [`slice`] contains a boolean value for each index even if it is a
+        /// separator. However it is guaranteed that all the separator indices will be false.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// use requestty::Question;
+        ///
+        /// let order_select = Question::order_select("home_tasks")
+        ///     //...
+        ///     .validate(|tasks, previous_answers| {
+        ///         if tasks[0].text() == "Make the bed" {
+        ///             Err("You have to make the bed first".to_string())
+        ///         } else {
+        ///             Ok(())
+        ///         }
+        ///     })
+        ///     //...
+        ///     .build();
+        /// ```
```

Comment:
Remove the "NOTE", and like the tests above, the then and else block are swapped. The condition is checking that the first one is "Make the bed", but if it is, it is returning an error that the first one should be to make the bed.